### PR TITLE
[testing workflows] Include more paths in changes lists

### LINKED ISSUE #50374: PR checks should run for changes to woocommerce.php
`plugins/woocommerce/woocommerce.php` is not included in any of the `changes` list in the CI config, so updates to this file will not trigger any tests.
We should update the config to include the file.

Context: p1722874815063899/1722856955.408389-slack-C01DT6U03HC

### PR DESCRIPTION
**Pull request by adimoldovan**
### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/50374

Update the changes lists for some test jobs so they get triggered in CI.

### How to test the changes in this Pull Request:

1. Create a branch from this PRs branch (`ci/include-woocommercephp-in-changes`)
2. Change a relevant file
3. Run `pnpm utils ci-jobs -r ci/include-woocommercephp-in-changes`
4. Check the output

### CODE REVIEW

File: plugins/woocommerce/package.json [hunk 1]
```diff
@@ -122,7 +116,10 @@
 						"tests/php/**/*.php",
 						"tests/legacy/unit-tests/**/*.php",
 						"tests/unit-tests/**/*.php",
-						".wp-env.json"
+						".wp-env.json",
+						"woocommerce.php",
```
Comment (naman03malhotra) [inline plugins/woocommerce/package.json]: I was wondering if there is a case, where we won't be triggering CI for a change in `*.php` file across the project?

Pehaps maybe we can consider just tirggering CI for `**/*.php` and remove the rest of the paths?

Comment (adimoldovan) [reply plugins/woocommerce/package.json]: I agree it makes sense to simply use `**/*.php` for PHP unit tests jobs. There are some php files in `tests/e2e-pw/bin` that should not trigger unit tests, but I images those are rarely updated.

Comment (adimoldovan) [reply plugins/woocommerce/package.json]: Updated with [f3f23ea](https://github.com/woocommerce/woocommerce/pull/50399/commits/f3f23eafee37cc1701aa7aae5818374954da6ad4)

**Review by naman03malhotra — APPROVED** [review]
Tested with this PR: https://github.com/woocommerce/woocommerce/pull/50551

Works as expected, thank you for quick turn around on this one 🙇

**Pull request merged**